### PR TITLE
Add set_all_channels() for setting on, off counter values and full-on and full-off in one transaction

### PR DESCRIPTION
**Pull request by mhthies**
Follow-up and alternative solution to PR #17. See discussion over there for detailed description of the problem and advantages and disadvantages of the solution approach.

### CODE REVIEW

File: src/channels.rs [hunk 1]
```diff
@@ -118,6 +118,38 @@ where
...
+    ) -> Result<(), Error<E>> {
+        let mut data = [0; 65];
+        data[0] = Register::C0_ON_L;
+        for (i, channel_value) in values.iter().enumerate() {
+            if channel_value.on > 4095 || channel_value.off > 4095 {
+                return Err(Error::InvalidInputData);
+            }
+            data[i * 4 + 1] = channel_value.on as u8;
+            data[i * 4 + 2] =
+                (channel_value.on >> 8) as u8 | (0b0001_0000 * channel_value.full_on as u8);
```
Comment (eldruin) [inline src/channels.rs:142]: Could you create a local constant like `const FULL_ON_OFF: u8 = 0b0001_0000;` for this?

File: tests/channels.rs [hunk 2]
```diff
@@ -360,3 +378,181 @@ fn can_set_all_on_off() {
...
+                4,
+                4,
+                3,
+                2,
+                5,
+                4,
+                4,
+                2,
+                6,
+                4 + 16,
```
Comment (eldruin) [inline tests/channels.rs:446]: Could you use a local constant for the full on/off like above here as well for the value `16`?

**Review by eldruin — COMMENTED** [review]
Great, thank you!
Could you also add an entry to the changelog?

**Review by eldruin — COMMENTED** [review]
Sorry, could you also briefly explain this function in the readme/lib.rs like the rest?

**Review by eldruin — COMMENTED** [review]
Could you run `cargo fmt` again?

**Comment by mhthies** [comment]
Done. Sorry for not checking this. I ran `cargo doc`, but I didn't consider that `cargo fmt` might dislike a random extra space in the comment.

**Comment by eldruin** [comment]
I have published this in [version 1.0](https://github.com/eldruin/pwm-pca9685-rs/releases/tag/v1.0.0). Thanks!

**Pull request merged**